jib-cli/src/main/java/com/google/cloud/tools/jib/cli/jar/JarFiles.java

```diff
@@ -30,20 +46,55 @@
    * Generates a {@link JibContainerBuilder} from contents of a jar file.
    *
    * @param processor jar processor
+   * @param jarOptions jar cli options
+   * @param commonCliOptions common cli options
+   * @param logger console logger
    * @return JibContainerBuilder
    * @throws IOException if I/O error occurs when opening the jar file or if temporary directory
    *     provided doesn't exist
    * @throws InvalidImageReferenceException if the base image reference is invalid
    */
-  public static JibContainerBuilder toJibContainerBuilder(JarProcessor processor)
+  public static JibContainerBuilder toJibContainerBuilder(
+      JarProcessor processor,
+      Jar jarOptions,
+      CommonCliOptions commonCliOptions,
+      ConsoleLogger logger)
       throws IOException, InvalidImageReferenceException {
 
-    // Use distroless as the base image.
-    JibContainerBuilder containerBuilder = Jib.from("gcr.io/distroless/java");
+    // Use distroless as the default base image.
+    JibContainerBuilder containerBuilder =
+        jarOptions.getFrom().isPresent()
+            ? createJibContainerBuilder(jarOptions.getFrom().get(), commonCliOptions, logger)
+            : Jib.from("gcr.io/distroless/java");
 
     List<FileEntriesLayer> layers = processor.createLayers();
     List<String> entrypoint = processor.computeEntrypoint();
     containerBuilder.setEntrypoint(entrypoint).setFileEntriesLayers(layers);
     return containerBuilder;
   }
+
+  private static JibContainerBuilder createJibContainerBuilder(
+      String from, CommonCliOptions commonCliOptions, ConsoleLogger logger)
+      throws InvalidImageReferenceException, FileNotFoundException {
+    String baseImageReference = from;
```

Comment:
It's unfortunate that the code is duplicated here and in `BuildFiles`. @loosebazooka thoughts?

Reply:
Yeah I think the two methods can be merged into one with the signature:

```
(String image, @Nullable List<Platform> platforms, CommonCliOptions commonCliOptions, ConsoleLogger logger)
```

We could put it in a new file `ContainerBuilders.java` for now next to `Containerizers` (in `com.google.cloud.tools.jib.cli`). We can move it around later if it makes more sense?

Reply:
Sounds like a good idea. I'll make this change in this PR since it doesn't look like a major refactoring?

Reply:
sgtm